src/memory.rs

```diff
@@ -184,6 +279,88 @@ impl Mul<f32> for MemoryQuantity {
     }
 }
 
+impl Div<f32> for MemoryQuantity {
+    type Output = Self;
+
+    fn div(self, rhs: f32) -> Self::Output {
+        self * (1. / rhs)
+    }
+}
+
+impl Sub<MemoryQuantity> for MemoryQuantity {
+    type Output = MemoryQuantity;
+
+    fn sub(self, rhs: MemoryQuantity) -> Self::Output {
+        if rhs.unit == self.unit {
+            MemoryQuantity {
+                value: self.value - rhs.value,
+                unit: self.unit,
+            }
+        } else if rhs.unit < self.unit {
+            MemoryQuantity {
+                value: self.scale_to(rhs.unit).value - rhs.value,
+                unit: rhs.unit,
+            }
+        } else {
+            MemoryQuantity {
+                value: self.value - rhs.scale_to(self.unit).value,
+                unit: self.unit,
+            }
+        }
+    }
+}
+
+impl Add<MemoryQuantity> for MemoryQuantity {
+    type Output = MemoryQuantity;
+
+    fn add(self, rhs: MemoryQuantity) -> Self::Output {
+        if rhs.unit == self.unit {
+            MemoryQuantity {
+                value: self.value + rhs.value,
+                unit: self.unit,
+            }
+        } else if rhs.unit < self.unit {
+            MemoryQuantity {
+                value: self.scale_to(rhs.unit).value + rhs.value,
+                unit: rhs.unit,
+            }
+        } else {
+            MemoryQuantity {
+                value: self.value + rhs.scale_to(self.unit).value,
+                unit: self.unit,
+            }
+        }
+    }
```

Comment:
I think its counter intuitive to eventually have the `self` unit changed. I think this can be shortend to (same for substract):
```suggestion
    fn add(self, rhs: MemoryQuantity) -> Self::Output {
        MemoryQuantity {
            value: self.value + rhs.scale_to(self.unit).value,
            unit: self.unit,
        }
    }
```